src/Raven.Client/Documents/Session/AsyncSessionDocumentTimeSeries.cs

```diff
@@ -385,7 +385,11 @@ private void RegisterIncludes(TimeSeriesDetails details)
                 // add current range from cache to the merged list.
                 // in order to avoid duplication, skip first item in range if needed
 
-                mergedValues.AddRange(ranges[i].Entries.Skip(mergedValues.Count == 0 ? 0 : 1));
+                bool shouldSkip = false;
+                if (mergedValues.Count > 0)
+                    shouldSkip = ranges[i].Entries[0].Timestamp == mergedValues[mergedValues.Count - 1].Timestamp;
```

Comment:
Can you explain me this change?

Reply:
Sure. For example, take a look at `RemovingTimeSeriesEntryShouldAffectCache3` test, there is one range left in cache (with `timestamp= "2023-02-15T09:01:43.6200000Z";` `value = 10`), while the last entry in `mergedValues = "2023-02-14T09:01:43.6200000Z"` - thus, before the fix, we skipped the value existing in cache and get incomplete values...